### PR TITLE
Trajectory Following Integration

### PR DESCRIPTION
**Pull request by StephenWelch**
- Fixes more timing problems
- Working trajectory from front left cargo ship hatch to loading station

### CODE REVIEW

File: robot/src/main/java/us/ilite/lib/drivers/NavX.java [hunk 1]
```diff
@@ -16,7 +16,8 @@
 	
  	public NavX(Port pPort){
  	  super(kCollisionGains);
- 	  iahrs = new AHRS(pPort);
+ 	  iahrs = new AHRS(pPort, AHRS.SerialDataType.kProcessedData, (byte)200);
```
Comment (oconnelc) [inline robot/src/main/java/us/ilite/lib/drivers/NavX.java]: Nitpick: do we want to make this a constant? It's unclear why 200 was selected.

File: robot/src/main/java/us/ilite/lib/drivers/NavX.java [hunk 2]
```diff
@@ -16,7 +16,8 @@
...
+ 	  System.out.println("NavX Update Rate: " + iahrs.getActualUpdateRate());
```
Comment (oconnelc) [inline robot/src/main/java/us/ilite/lib/drivers/NavX.java]: Recommend using the logger

File: robot/src/main/java/us/ilite/robot/auto/paths/FieldElementLocations.java [hunk 3]
```diff
@@ -12,7 +12,7 @@
     //------------------------------
 
     // Two middle hatches
-    public static final Translation2d kCargoShipMiddleLeftHatch = new Translation2d(150.0, 150.0);
+    public static final Translation2d kCargoShipMiddleLeftHatch = new Translation2d(18.5 * 12.0 , 12.0 * 12.0);
```
Comment (oconnelc) [inline robot/src/main/java/us/ilite/robot/auto/paths/FieldElementLocations.java:15]: What does this math mean? In other words: what does 18.5 x 12 and 12 x 12 mean?

File: robot/src/main/java/us/ilite/robot/auto/paths/FieldElementLocations.java [hunk 4]
```diff
@@ -23,7 +23,7 @@
     //-----------------
     // Loading Station
     //-----------------
-    public static final Translation2d kLoadingStation = new Translation2d();
+    public static final Translation2d kLoadingStation = new Translation2d(0.0, 2.0 * 12.0);
```
Comment (oconnelc) [inline robot/src/main/java/us/ilite/robot/auto/paths/FieldElementLocations.java:26]: What is it 2.0 * 12, why not just hardcode to 24.0?

File: robot/src/main/java/us/ilite/robot/auto/paths/RobotDimensions.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package us.ilite.robot.auto.paths;
+
+public class RobotDimensions {
+
+    public static final double kBackToCenter = 18.6;
```
Comment (oconnelc) [inline robot/src/main/java/us/ilite/robot/auto/paths/RobotDimensions.java:8]: Recommend documenting units in javadoc and/or name of variable.

File: robot/src/main/java/us/ilite/robot/auto/paths/StartingPoses.java [hunk 6]
```diff
@@ -6,6 +6,6 @@
 public class StartingPoses {
 
     public static final Pose2d kMiddleStart = new Pose2d();
-    public static final Pose2d kSideStart = new Pose2d(36.0, 50.0, Rotation2d.fromDegrees(0.0));
+    public static final Pose2d kSideStart = new Pose2d(4.0 * 12.0 + RobotDimensions.kBackToCenter, 11.5 * 12.0 + RobotDimensions.kSideToCenter, Rotation2d.fromDegrees(0.0));
```
Comment (oconnelc) [inline robot/src/main/java/us/ilite/robot/auto/paths/StartingPoses.java:18]: Please document what this math means.

File: robot/src/main/java/us/ilite/robot/auto/paths/TestAuto.java [hunk 7]
```diff
@@ -3,11 +3,14 @@
 import com.team254.lib.geometry.Pose2d;
 import com.team254.lib.geometry.Rotation2d;
 
+import java.lang.reflect.Field;
```
Comment (oconnelc) [inline robot/src/main/java/us/ilite/robot/auto/paths/TestAuto.java]: This import is not used, please remove.

File: robot/src/main/java/us/ilite/robot/commands/FollowTrajectory.java [hunk 8]
```diff
@@ -1,13 +1,19 @@
...
 import com.team254.lib.geometry.Pose2dWithCurvature;
+import com.team254.lib.geometry.Translation2d;
 import com.team254.lib.trajectory.Trajectory;
 import com.team254.lib.trajectory.timing.TimedState;
 import edu.wpi.first.wpilibj.Timer;
 import us.ilite.robot.modules.Drive;
 
 public class FollowTrajectory implements ICommand {
 
+    private final ILog mLog = Logger.createLog(FollowTrajectory.class);
```
Comment (oconnelc) [inline robot/src/main/java/us/ilite/robot/commands/FollowTrajectory.java:15]: Nitpick: make this static.

**Review by oconnelc — CHANGES_REQUESTED** [review]
Mostly nitpicks, but some thing should be cleaned up/documented.

File: robot/src/main/java/us/ilite/robot/auto/paths/FieldElementLocations.java (hunk 4 above)
Comment (StephenWelch) [reply robot/src/main/java/us/ilite/robot/auto/paths/FieldElementLocations.java:26]: I was trying to be consistent - nearly all of my field measurements are converted from feet -> inches

File: robot/src/main/java/us/ilite/robot/auto/paths/FieldElementLocations.java (hunk 3 above)
Comment (StephenWelch) [reply robot/src/main/java/us/ilite/robot/auto/paths/FieldElementLocations.java:15]: This is an (x, y) position

File: common/src/main/java/us/ilite/common/config/SystemSettings.java [hunk 9]
```diff
@@ -68,7 +68,12 @@
     public static double 	DRIVETRAIN_TURN_CIRCUMFERENCE = kDriveEffectiveWheelbase * Math.PI;
     public static double	DRIVETRAIN_INCHES_PER_DEGREE = DRIVETRAIN_TURN_CIRCUMFERENCE / 360.0;
     public static double	DRIVETRAIN_WHEEL_TURNS_PER_DEGREE = DRIVETRAIN_INCHES_PER_DEGREE / kDriveWheelDiameterInches;
-    public static double kDriveCollisionThreshold = 0.0;
+
+    // =============================================================================
+    // IMU Constants
+    // =============================================================================
+    public static double kGyroCollisionThreshold = 0.0;
+    public static int kGyroUpdateRate = 200.0;
```
Comment (oconnelc) [inline common/src/main/java/us/ilite/common/config/SystemSettings.java]: This is causing a compile error: 200.0 is a double not an int.

**Review by oconnelc — CHANGES_REQUESTED** [review]

**Review by DanielTolbert — DISMISSED** [review]

**Review by oconnelc — APPROVED** [review]

**Review by DanielTolbert — APPROVED** [review]

**Pull request merged**